src/dotnet/commands/dotnet-restore3/Program.cs

```diff
@@ -97,6 +103,13 @@ public static int Run(string[] args)
                     msbuildArgs.Add($"/p:RestoreIgnoreFailedSources=true");
                 }
 
+                // By default restore is recursive, if set to false turn it off.
```

Comment:
I agree with what @eerhardt said. CLI UX-es have to have consistency between options since that is the only way to actually build a sane CLI toolset. 

I personally don't think that  `--no-dependencies`is a perfect name. Maybe we should call it something else, like `--no-project-references` or similar? After all, in MSBuild world, everything is a "reference"...